### PR TITLE
Add a bunch of compoud query fixtures, with dubious analysis

### PR DESCRIPTION
**Pull request by crisptrutski**
Closes https://github.com/metabase/metabase/issues/42585

I think these case a wide enough net for the next spurt of development. I'll leave fixing the actual result values to the various PRs fixing the explanatory TODOs, as I haven't finalized the `should=` style macros yet.

The main gap is handling columns that come from sentinel tables, but that'll require some API changes and I'll leave them for now.

### CODE REVIEW

File: test/macaw/core_test.clj [hunk 1]
```diff
@@ -503,20 +507,88 @@
     (components (query-fixture :snowflakelet))
     (components (query-fixture :snowflake))))
 
-(defn sorted-vec
+(defn sorted
   "A transformation to help write tests, where hawk would face limitations on predicates within sets."
   [element-set]
-  (sort-by (comp (juxt :schema :table :column :scope) :component) element-set))
+  (sort-by (comp (juxt :schema :table :column :scope) #(:component % %)) element-set))
```
Comment (crisptrutski) [inline test/macaw/core_test.clj:513]: Note: this is now agnostic to whether we've stripped the scope or not.

**Comment by crisptrutski** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/metabase/macaw/56?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#56** <a href="https://app.graphite.dev/github/pr/metabase/macaw/56?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#54** <a href="https://app.graphite.dev/github/pr/metabase/macaw/54?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @crisptrutski and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: test/macaw/core_test.clj [hunk 2]
```diff
@@ -14,14 +14,15 @@
 
 (set! *warn-on-reflection* true)
 
-(defn- and*
-  [x y]
-  (and x y))
+(defn- non-empty-and-truthy [xs]
```
Comment (tsmacdonald) [inline test/macaw/core_test.clj:17]: I sometimes see this spelled `present?`, but :shrug:

File: test/macaw/core_test.clj [hunk 3]
```diff
@@ -521,20 +526,92 @@ from foo")
...
+           {:component {#_#_:table "a", :column "x"}, :scope ["SELECT" (=?/same :subselect-2)]}
            {:component {:table "b", :column "x"}, :scope ["SUB_SELECT" (=?/same :top-level)]}
            {:component {:table "c", :column "x"}, :scope ["SUB_SELECT" (=?/same :top-level)]}]
-          (sorted-vec (contexts->scopes (:columns (components (query-fixture :duplicate-scopes))))))))
+          (sorted (contexts->scopes (:columns (components (query-fixture :duplicate-scopes))))))))
+
+(deftest count-field-test
+  (testing "COUNT(*) does not actually read any columns"
+    (is (empty? (columns "SELECT COUNT(*) FROM users")))
+    #_(is (false? (has-wildcard? "SELECT COUNT(*) FROM users")))
```
Comment (tsmacdonald) [inline test/macaw/core_test.clj:547]: aww, this fails?

File: test/macaw/core_test.clj [hunk 4]
```diff
@@ -521,20 +526,92 @@ from foo")
...
+  (testing "We do care about deeply referenced fields in a COUNT however"
+    (is (= #{{:table "users" :column "id"}}
+           (columns "SELECT COUNT(DISTINCT(id)) FROM users")))))
+
+(deftest compound-subselect-test
+  ;; TODO These first three entries should track what they're derived from, so we can filter them from query fields.
+  (is (= [{:column "average_salary", :alias "avg_salary"}
+          {:column "department", :alias "department_name"}
+          {:column "department"}
+          ;; TODO This doesn't belong here, as the identifier does not relate to any collumn
```
Comment (tsmacdonald) [inline test/macaw/core_test.clj:565]:
```suggestion
          ;; TODO This doesn't belong here, as the identifier does not relate to any column
```

**Review by tsmacdonald — APPROVED** [review]
Big step in a good direction

File: test/macaw/core_test.clj (hunk 3 above)
Comment (crisptrutski) [reply test/macaw/core_test.clj:547]: Unfortunately yes, but will fix shortly.

File: test/macaw/core_test.clj (hunk 2 above)
Comment (crisptrutski) [reply test/macaw/core_test.clj:17]: Yeah I also like that Rails-y terminology, will make the change next time I pass through.

**Pull request merged**